### ISSUE TITLE
asm syntax doesn't take comptime strings

### ISSUE THREAD

**Issue by daurnimator** [ziglang/zig#2430]
Use case: while working on #2426, I tried to do:
```zig
pub fn ceilPowerOfTwo3(comptime T: type, value: T) T {
    asm volatile(("# LLVM-MCA-BEGIN ceilPowerOfTwo3(" ++ @typeName(T) ++ ", x)"));
```
Example error: `error: expected token 'StringLiteral', found '('`

So that I'd get an [llvm-mca](https://llvm.org/docs/CommandGuide/llvm-mca.html#using-markers-to-analyze-specific-code-blocks) report for each different type that `ceilPowerOfTwo3` was called with.

-----------------

Bonus bad error message to fix:
```zig
ceil.zig:10:54: error: expected token ')', found '++'
    asm volatile("# LLVM-MCA-BEGIN ceilPowerOfTwo3(" ++ @typeName(T) ++ ", x)");
```

**Comment by vegecode** [comment]
Another use case is more faithfully translating the preprocessor defines for the assembly functions in compiler-rt.

**Comment by andrewrk** [comment]
Fixed by #3829, landed in 26c8930b9594ab320b0e3add682e99063491142e
